### ISSUE TITLE
Unable to deploy on OpenShift

### ISSUE THREAD

**Issue by mfrancis95** [nginxinc/docker-nginx-unprivileged#5]
See #4.

**Comment by mfrancis95** [comment]
I've tried all four tags and they give the same permission issue.

**Comment by pvanagtmaal** [comment]
I can confirm the issue on stable-alpine:

```
2018/09/10 13:32:57 [emerg] 1#1: mkdir() "/var/cache/nginx/proxy_temp" failed (13: Permission denied)
nginx: [emerg] mkdir() "/var/cache/nginx/proxy_temp" failed (13: Permission denied)
```

**Comment by alessfg** [comment]
Hey all!

I just pushed an update both to GitHub and Docker that should hopefully address these issues. Let me know if it works!

**Comment by pvanagtmaal** [comment]
Just confirmed: both mainline and stable run! Thank you!

**Comment by alessfg** [comment]
Great!

**Comment by KevinWMatthews** [comment]
@mfrancis95 and @pvanagtmaal , do you know if OpenShift runs the container as a particular user? If the container is run as `nginx` or `root`, it seems like you shouldn't have had an issue.

I'm not sure how the OpenShift system works, but one can query a standalone, running container using:
```
$ docker exec -it <container_name> whoami
$ docker exec -it <container_name> id
```

My results before the latest commit are:
```
$ docker exec -it <container_name> whoami
nginx
$ docker exec -it <container_name> id
uid=100(nginx) gid=101(nginx) groups=101(nginx)
```

My results after the latest commit are:
```
$ docker exec -it <container_name> whoami
root
$ docker exec -it <container_name> id
uid=0(root) gid=0(root) groups=0(root),1(bin),2(daemon),3(sys),4(adm),6(disk),10(wheel),11(floppy),20(dialout),26(tape),27(video)
```

Our unprivileged container is **running as root**!

The latest commit,
[d68ed1](https://github.com/nginxinc/docker-nginx-unprivileged/commit/d68ed10bd0ddc54503027efc66d0d5b141755ff9), simply runs as the root user. If we're trying to run an unprivileged container, this has directly undone that.

What happens if you run the [privileged nginx](https://github.com/nginxinc/docker-nginx) container in OpenShift?

Thanks!

**Comment by alessfg** [comment]
Reopening this issue because it's clear that further R&D has to be done on how to run an NGINX container in OpenShift following best practices.

**Comment by pvanagtmaal** [comment]
@KevinWMatthews OpenShift runs as a random user but with the `root` group, see this:

[https://docs.openshift.com/container-platform/3.3/creating_images/guidelines.html#openshift-container-platform-specific-guidelines](https://docs.openshift.com/container-platform/3.3/creating_images/guidelines.html#openshift-container-platform-specific-guidelines)

This translates to the following in my container:

```
$ whoami
whoami: cannot find name for user ID 1000210000
$ id
uid=1000210000 gid=0(root) groups=0(root),1000210000
```

**Comment by KevinWMatthews** [comment]
@pvanagtmaal Awesome, thanks for your feedback. 

You're welcome to weigh in on #10 and #8 - the discussion is shifting to 10 in particular.

**Comment by KevinWMatthews** [comment]
@mfrancis95 and @pvanagtmaal , did this get fixed by #10 ? I think/hope so.

**Comment by alessfg** [comment]
Fixed by #10
